flowmachine/tests/test_per_subscriber_aggregate.py

```diff
@@ -0,0 +1,63 @@
+from flowmachine.features.subscriber.per_subscriber_aggregate import (
+    PerSubscriberAggregate,
+    agg_methods,
+)
+from flowmachine.features.subscriber.subscriber_call_durations import (
+    PerLocationSubscriberCallDurations,
+)
+from flowmachine.core.dummy_query import DummyQuery
+
+import pytest
+
+
+@pytest.fixture()
+def per_location_query(flowmachine_connect):
+    sq = PerLocationSubscriberCallDurations("2016-01-01", "2016-01-02")
+    yield sq
+
+
+class NonSubQuery(DummyQuery):
+    @property
+    def column_names(self):
+        return ["value"]
+
+
+@pytest.mark.parametrize("agg_method", agg_methods)
+def test_aggregates(get_dataframe, agg_method, per_location_query):
+
+    psa = PerSubscriberAggregate(
+        subscriber_query=per_location_query,
+        agg_column="value",
+        agg_method=agg_method,
+    )
+    # per_location_query is 608 rows
+    assert (
+        len(get_dataframe(psa))
+        == get_dataframe(per_location_query).subscriber.nunique()
+    )
```

Comment:
One thing this won't catch is if an incorrect aggregation method is used (e.g. if the 'agg_method' param gets ignored, and "avg" is always used by mistake). Would be good to have a test to catch that. Where possible I usually try to avoid golden value tests that explicitly depend on the test data, but this might be a situation where that's the most pragmatic option.